### PR TITLE
Rename variables ending in "_"

### PR DESCRIPTION
**Pull request by fergyfresh**
Attempting to satisfy #133

I'm about 1/2 way there and just wanted to throw this up cause there's alot to look at so far. 

All the tests pass and a few things needed to be swapped around `alias_`, `end_` and pretty much all other variables that I will come across that are reserved words will need to be changed slightly.

### CODE REVIEW

**Comment by teeparham** [comment]
This is looking great. Thanks for pushing your work so far. 👍

**Comment by fergyfresh** [comment]
Finally finished up this bad boy. I did a once over and fixed some stragglers and then I did a twice over and didn't find any more. We might be good on this finally.

File: lib/rgeo/geos/ffi_factory.rb [hunk 1]
```diff
@@ -149,88 +149,88 @@ def marshal_dump # :nodoc:
...
-            proj4_ = CoordSys::Proj4.create(proj4_data_.to_s)
+            proj4 = CoordSys::Proj4.create(proj4_data.to_s)
           end
         else
-          proj4_ = nil
+          proj4 = nil
         end
-        if (coord_sys_data_ = coder_["cs"])
-          coord_sys_ = CoordSys::CS.create_from_wkt(coord_sys_data_.to_s)
+        if (coord_sysdata = coder["cs"])
```
Comment (teeparham) [inline lib/rgeo/geos/ffi_factory.rb]: `coord_sys_data`

File: lib/rgeo/geos/ffi_feature_methods.rb [hunk 2]
```diff
@@ -406,8 +406,8 @@ def centroid
         @factory._wrap_fg_geom(@fg_geom.centroid, FFIPointImpl)
       end
 
-      def point_on_surface
-        @factory._wrap_fg_geom(@fg_geom.point_on_surface, FFIPointImpl)
+      def point_onsurface
```
Comment (teeparham) [inline lib/rgeo/geos/ffi_feature_methods.rb]: `point_on_surface`

File: lib/rgeo/geos/interface.rb [hunk 3]
```diff
@@ -29,30 +29,30 @@ def supported?
       # Returns true if the given feature is a CAPI GEOS feature, or if
       # the given factory is a CAPI GEOS factory.
 
-      def is_capi_geos?(object_)
+      def is_capi_geos?(obj)
```
Comment (teeparham) [inline lib/rgeo/geos/interface.rb]: `object`. (it's not a keyword, so it's ok as a variable name).  The comments in this file were changed too.

File: lib/rgeo/geos/zm_feature_methods.rb [hunk 4]
```diff
@@ -337,7 +337,7 @@ def centroid
         @factory._create_feature(ZMPointImpl, @zgeometry.centroid, @mgeometry.centroid)
       end
 
-      def point_on_surface
+      def point_onsurface
```
Comment (teeparham) [inline lib/rgeo/geos/zm_feature_methods.rb]: `point_on_surface`

**Comment by fergyfresh** [comment]
Thanks for the quick feedback, fixed all of your immediate issues.

**Comment by teeparham** [comment]
Thank you! This should make all the code easier to look at. And it should make the gem a bit smaller too. 🎉 🎉 🎉

**Pull request merged**